03 - CSS Variables/index.html

```diff
@@ -42,9 +48,30 @@ <h2>Update CSS Variables with <span class='hl'>JS</span></h2>
     input {
       width: 100px;
     }
+
+    img {
+      padding: var(--spacing);
+      filter: blur(var(--blur));
+      background-color: var(--base);
+    }
+
+    h2 .hl {
+      color: var(--base);
+    }
   </style>
 
   <script>
+    let inputs = document.querySelectorAll( ".controls input" );
+    inputs.forEach(input => input.addEventListener( "change", updateValue ));
+
+    function updateValue() {
+      if( isNaN(this.value) ) {
+        document.documentElement.style.setProperty( `--${this.name}`, `${this.value}` );
+      } else {
+        document.documentElement.style.setProperty( `--${this.name}`, `${this.value}px` );
+      }
+    }
```

Comment:
This function can be optimized by using a ternary operator and calling the `setProperty` function once. Consider something like:

```
function updateValue() {
  const value = isNaN(this.value) ? this.value : `${this.value}px`;
  document.documentElement.style.setProperty( `--${this.name}`, value );
}
```

Reply:
Oh right, this is much more simpler.

My original plan was using ternary operator as well, but the difference is that mine was like:

```
function updateValue() {
  isNaN(this.value) ? 
      document.documentElement.style.setProperty( `--${this.name}`, `${this.value}` ) :
      document.documentElement.style.setProperty( `--${this.name}`, `${this.value}` );
}
```

And I'm still not sure on what extent a good and bad practice is when it comes to ternary operators, which in this case my code was long so I switched it to a regular if else block.